kernel/fde/fde_test.go

```diff
@@ -505,11 +506,32 @@ func (s *fdeSuite) TestRevealErr(c *C) {
 		SealedKey: sealedKey,
 	}
 	_, err := fde.Reveal(&p)
-	c.Assert(err, ErrorMatches, `(?s)cannot run fde-reveal-key "reveal": 
------
-failed
-service result: exit-code
------`)
+	c.Assert(err, ErrorMatches, `(?s)cannot run fde-reveal-key "reveal": failed`)
+
+	// several parameters for systemd-run include the GlobalRootDir, which has
+	// been dynamically generated with a random name at the beginning of this
+	// test suite.
+	// In order to get a predictable output, replace all such occurrences in
+	// the parameters with "/ROOT".
+	calls := mockSystemdRun.Calls()
+	for _, call := range calls {
+		for j, param := range call {
+			call[j] = strings.ReplaceAll(param, dirs.GlobalRootDir, "/ROOT")
+		}
+	}
+	c.Check(calls, DeepEquals, [][]string{
+		{
+			"systemd-run", "--collect", "--service-type=exec", "--quiet",
+			"--property=RuntimeMaxSec=2m0s",
+			"--property=SystemCallFilter=~@mount",
+			"--property=StandardInput=file:/ROOT/run/fde-reveal-key/fde-reveal-key.stdin",
```

Comment:
the slightly more idiomatic way (well less idiomatic than historical, not sure what other projects do) we do this in other places in the codebase is like this:
```suggestion
			fmt.Sprintf("--property=StandardInput=file:%s/run/fde-reveal-key/fde-reveal-key.stdin", dirs.GlobalRootDir)
```
See for example the many many many tests which do this in cmd_initramfs_mounts_test.go and then try not to run away in terror from that part of the codebase :smile:

Reply:
Done!